client Open() without protocol panics when used in Txn

Not specifying the "http://" prefix in the db name panics the txnSender.
Probably should catch this as invalid input sometime earlier.

```
16:51 $ ./bank -num-accounts=100 -num-parallel-transfers=5 -db-name=root@localhost:8001
A simple program that keeps moving money between bank accounts.

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x4358290]

goroutine 1 [running]:
github.com/cockroachdb/cockroach/client.(*txnSender).Send(0xc208036ea0, 0x53a0890, 0xc20802abd0, 0x53a07c8, 0xc208078a20, 0x53a0820, 0xc208060300, 0x0, 0x0, 0x0)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/client/txn.go:53 +0xe0
github.com/cockroachdb/cockroach/client.(*DB).send(0xc208036ea0, 0xc20800be40, 0x1, 0x1, 0x0, 0x0)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/client/db.go:396 +0x3d8
github.com/cockroachdb/cockroach/client.(*Txn).send(0xc208036ea0, 0xc20800be40, 0x1, 0x1, 0x0, 0x0)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/client/txn.go:307 +0xa8
github.com/cockroachdb/cockroach/client.(*Txn).Run(0xc208036ea0, 0xc208066c00, 0x0, 0x0)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/client/txn.go:241 +0x8e
github.com/cockroachdb/cockroach/client.runOneResult(0x53a0868, 0xc208036ea0, 0xc208066c00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/client/db.go:883 +0x6b
github.com/cockroachdb/cockroach/client.(*Txn).Scan(0xc208036ea0, 0x487b260, 0xc20801ed20, 0x487b260, 0xc20801ed40, 0x64, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/client/txn.go:201 +0x101
main.func·004(0xc208036ea0, 0x0, 0x0)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/examples/bank/bank.go:171 +0x19b
github.com/cockroachdb/cockroach/client.func·007(0x538e000, 0x0, 0x0)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/client/txn.go:266 +0x80
github.com/cockroachdb/cockroach/util/retry.WithBackoff(0xc20802b520, 0xb, 0x2faf080, 0x12a05f200, 0x4000000000000000, 0x0, 0x1, 0x0, 0xc2080a7c98, 0x0, ...)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/retry/retry.go:86 +0x9e
github.com/cockroachdb/cockroach/client.(*Txn).exec(0xc208036ea0, 0xc2080a7e88, 0x0, 0x0)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/client/txn.go:287 +0xd3
github.com/cockroachdb/cockroach/client.(*DB).Txn(0xc2080102a0, 0xc2080a7e88, 0x0, 0x0)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/client/db.go:368 +0x75
main.(*Bank).initBankAccounts(0xc20801ed00, 0x3e8, 0x0)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/examples/bank/bank.go:200 +0x7a
main.main()
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/examples/bank/bank.go:262 +0x43c
```

--- a/client/db.go
+++ b/client/db.go
@@ -226,6 +226,10 @@ func Open(addr string, opts ...Option) (*DB, error) {
 	for _, opt := range opts {
 		opt(db)
 	}
+
+	if db.Sender == nil {
+		return nil, fmt.Errorf("\"%s\" no sender specified", addr)
+	}
 	return db, nil
 }
 

--- a/client/db_test.go
+++ b/client/db_test.go
@@ -264,6 +264,7 @@ func TestOpenArgs(t *testing.T) {
 		{"https://root@" + s.ServingAddr() + "?certs=test_certs", false},
 		{"https://" + s.ServingAddr() + "?certs=test_certs", false},
 		{"https://" + s.ServingAddr() + "?certs=foo", true},
+		{s.ServingAddr(), true},
 	}
 
 	for _, test := range testCases {

